src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5559,14 +5552,15 @@ public override BoundNode VisitConvertedStackAllocExpression(BoundConvertedStack
 
         public override BoundNode VisitDiscardExpression(BoundDiscardExpression node)
         {
-            SetResult(node);
+            var result = TypeSymbolWithAnnotations.Create(node.Type);
+            SetResult(result, result);
```

Comment:
>SetResult(result, result); [](start = 12, length = 26)

Is this an l-value only?

Reply:
Yes, discards can only be written to, but cannot be read from.
But for IDE, you should be able to hover over it and see its type, so I think it's ok to set for the RValueType and the LValueType.

---
In reply to: [258673009](https://github.com/dotnet/roslyn/pull/33447#discussion_r258673009) [](ancestors = 258673009)